### PR TITLE
Bumped gRPC version to 1.3.0

### LINKED ISSUE #51099: gRPC update causing failure of API calls with large responses
For e.g the density test started failing on all our kubemark jobs that are big (i.e on >= 100 nodes):
- https://k8s-testgrid.appspot.com/sig-scalability#kubemark-100
- https://k8s-testgrid.appspot.com/sig-scalability#kubemark-500

And it clearly seems to be a regression caused by this PR (from the diff) - https://github.com/kubernetes/kubernetes/pull/50531 - which updated gRPC vendor code. The error we're seeing is the following:

```
Expected error:
    <*errors.StatusError | 0xc42d8c0900>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "rpc error: code = ResourceExhausted desc = grpc: received message larger than max (26710706 vs. 4194304)",
            Reason: "",
            Details: nil,
            Code: 500,
        },
    }
    rpc error: code = ResourceExhausted desc = grpc: received message larger than max (26710706 vs. 4194304)
not to have occurred
```

It's coming from this line of the test - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scalability/density.go#L232 - where we list all pods across all namespaces. Seems like the MTU for gRPC calls changed as a result of the update.

cc @kubernetes/sig-scalability-bugs @RenaudWasTaken @wojtek-t

### PR DESCRIPTION
**Pull request by RenaudWasTaken**
**What this PR does / why we need it**:

This PR bumps down the version of the vendored version of gRPC from v1.5.1 to v1.3.0
This is needed as part of the Device Plugin API where we expect client and server to use the Keep alive feature in order to detect an error.

Unfortunately I had to also bump the version of `golang.org/x/text` and `golang.org/x/net`.

- Design document: kubernetes/community#695
- PR tracking: [kubernetes/features#368](https://github.com/kubernetes/features/issues/368#issuecomment-321625420)

**Which issue this PR fixes**: fixes #51099
Which was caused by my previous PR updating to 1.5.1

**Special notes for your reviewer**:
@vishh @jiayingz @shyamjvs

**Release note**:
```
Bumped gRPC to v1.3.0
```

### CODE REVIEW

**Comment by RenaudWasTaken** [comment]
Forgot to update the staging Godeps that's done now

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1171,127 +1171,127 @@
 		},
 		{
 			"ImportPath": "github.com/gogo/protobuf/gogoproto",
-			"Comment": "v0.4-3-gc0656edd",
+			"Comment": "v0.4-3-gc0656ed",
```
Comment (vishh) [inline Godeps/Godeps.json]: What version is this? Is there no official release tag that we can use?

Comment (RenaudWasTaken) [inline Godeps/Godeps.json]: I agree that this is something that should be addressed however I don't believe this is the scope of this PR as neither this or the previous one changed the commit number.

**Comment by RenaudWasTaken** [comment]
@shyamjvs @wojtek-t @jbeda @lavalamp could you PTAL :) ?
All tests are passing and this should fix #51099 as per @xiang90's recommendations

**Comment by xiang90** [comment]
@shyamjvs can you do a stress test on this branch to confirm everything is OK?

**Comment by shyamjvs** [comment]
@RenaudWasTaken I'll test it on 100-node cluster - Can you please rebase before that?

**Comment by RenaudWasTaken** [comment]
@shyamjvs should be good now :)

**Comment by shyamjvs** [comment]
I'm on it.

**Comment by shyamjvs** [comment]
/lgtm

I tested it locally on 100-node cluster and it works now:

```
Ran 2 of 677 Specs in 2108.795 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 675 Skipped PASS

Ginkgo ran 1 suite in 35m9.89029736s
Test Suite Passed
2017/08/23 14:34:01 util.go:133: Step './hack/ginkgo-e2e.sh --ginkgo.focus=\[Feature:Performance\]' finished in 35m11.628136112s
```

**Comment by shyamjvs** [comment]
Thanks for fixing.
@wojtek-t Could you approve the PR?

**Comment by RenaudWasTaken** [comment]
Forgot once again to run `update-staging-godeps.sh`...
The verify job should now succeed

File: Godeps/Godeps.json (hunk 1 above)
Comment (vishh) [inline Godeps/Godeps.json]: There is an actual release - https://github.com/gogo/protobuf/releases/tag/v0.4
I hope a similar release would be available for other dependencies. 
I'd like to avoid vendoring in HEAD of dependencies.
@wojtek-t @shyamjvs since you are experts in this area, can you weigh in?

Comment (RenaudWasTaken) [inline Godeps/Godeps.json]: @vishh I'd like to avoid revendoring anything else than gRPC and it's dependecies (x/net and x/text).
I really think that doing so is outside the scope of this PR and would need to go through the whole re-testing on a 100 node cluster.

Let's keep the changes to the minimum to avoid breaking anything else :)

Comment (shyamjvs) [inline Godeps/Godeps.json]: I'm no expert here, but I'm seeing quite some packages in the file that don't seem to be from standard releases (e.g docker/engine-api, coreos/pkg). Apparently even the go-restful pkg.
IMO allowing only releases could become a serious limitation wrt feature velocity (i.e. wait for an upstream release before using it in your feature).

cc @thockin - the godep god :)
